src/api/stats.js

```diff
@@ -359,41 +362,46 @@ export function fetchLastImpressions(apiKey, baseQuery = {}, videoId) {
 
   return new Promise((resolve) => {
     api.fetchAnalytics('MIN', impressions).then((results) => {
-      const promises = [];
-      for (let i = 0; i < results.length; i++) {
-        promises.push(api.getImpression(results[i][0]));
+      Promise.map(results, result => {
+        return api.getImpression(result[0]).catch(() => null);
+      }, {
+        concurrency: CONCURRENT_REQUESTS_LIMIT
+      }).then(function (impressions) {
```

Comment:
let's not mix vanilla JS with es6, `function (impressions) {` would become `impressions => {`. easier to read and doesn't bloat the code as much ;)

Reply:
L370 same, please use `const` and `let` instead of `var`